DFC.ServiceTaxonomy.GraphSync/Queries/MatchSynonymsQuery.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections.Generic;
+using DFC.ServiceTaxonomy.Neo4j.Queries.Interfaces;
+using Neo4j.Driver;
+
+namespace DFC.ServiceTaxonomy.GraphSync.Queries
+{
+    public class MatchSynonymsQuery : IQuery<IRecord>
+    {
+        private string FirstNodeLabel { get; }
+        private string SecondNodeLabel { get; }
+        private string PropertyValue { get; }
+
+        private string RelationshipLabels { get; }
+
+        public MatchSynonymsQuery()
+        {
+            FirstNodeLabel = string.Empty;
+            SecondNodeLabel = string.Empty;
+            PropertyValue = string.Empty;
+            RelationshipLabels = string.Empty;
+        }
+
+        public MatchSynonymsQuery(string firstNodeLabel, string secondNodeLabel, string propertyValue, params string[] relationshipLabels)
+        {
+            FirstNodeLabel = firstNodeLabel;
+            SecondNodeLabel = secondNodeLabel;
+            PropertyValue = propertyValue;
+            RelationshipLabels = BuildRelationshipLabels(relationshipLabels);
+        }
+
+        private string BuildRelationshipLabels(string[] relationshipLabels)
+        {
+            for (var i = 0; i < relationshipLabels.Length; i++)
+                relationshipLabels[i] = ":" + relationshipLabels[i];
```

Comment:
This will leave a bunch of strings that'll need to be garbage collected.
Instead we could use `string.Join("|:", ...` and add an initial `:` before `{RelationshipLabels}` in the `Query` constructor.